java/query/src/main/java/sleeper/query/lambda/SqsLeafPartitionQueryProcessor.java

```diff
@@ -48,116 +43,61 @@
         sqsClient = builder.sqsClient;
         instanceProperties = builder.instanceProperties;
         tablePropertiesProvider = builder.tablePropertiesProvider;
         executorService = Executors.newFixedThreadPool(instanceProperties.getInt(EXECUTOR_POOL_THREADS));
         objectFactory = new ObjectFactory(instanceProperties, builder.s3Client, "/tmp");
         queryTracker = new DynamoDBQueryTracker(instanceProperties, builder.dynamoClient);
-        // The following Configuration is only used in StateStoreProvider for reading from S3 if the S3StateStore is used,
-        // so use the standard Configuration rather than the one for query lambdas which is specific to the table.
-        Configuration confForStateStore = HadoopConfigurationProvider.getConfigurationForLambdas(instanceProperties);
-        stateStoreProvider = new StateStoreProvider(builder.dynamoClient, instanceProperties, confForStateStore);
     }
 
     public static Builder builder() {
         return new Builder();
     }
 
-    public void processQuery(QueryOrLeafPartitionQuery query) {
+    public void processQuery(LeafPartitionQuery leafPartitionQuery) {
+        QueryOrLeafPartitionQuery query = new QueryOrLeafPartitionQuery(leafPartitionQuery);
         QueryStatusReportListeners queryTrackers = QueryStatusReportListeners.fromConfig(
-                query.getProcessingConfig().getStatusReportDestinations());
+                leafPartitionQuery.getProcessingConfig().getStatusReportDestinations());
         queryTrackers.add(queryTracker);
-        CloseableIterator<Record> results;
+
         try {
             TableProperties tableProperties = query.getTableProperties(tablePropertiesProvider);
-            if (query.isLeafQuery()) {
-                LeafPartitionQuery leafQuery = query.asLeafQuery();
-                queryTrackers.queryInProgress(leafQuery);
-                results = processLeafPartitionQuery(leafQuery, tableProperties);
-            } else {
-                Query parentQuery = query.asParentQuery();
-                queryTrackers.queryInProgress(parentQuery);
-                results = processRangeQuery(parentQuery, tableProperties, queryTrackers);
-            }
-            if (null != results) {
-                publishResults(results, query, tableProperties, queryTrackers);
-            }
-        } catch (StateStoreException | QueryException e) {
+            queryTrackers.queryInProgress(leafPartitionQuery);
+            Configuration conf = getConfiguration(tableProperties);
+            LeafPartitionQueryExecutor leafPartitionQueryExecutor = new LeafPartitionQueryExecutor(executorService, objectFactory, conf, tableProperties);
+            CloseableIterator<Record> results = leafPartitionQueryExecutor.getRecords(leafPartitionQuery);
+            publishResults(results, query, tableProperties, queryTrackers);
+        } catch (QueryException e) {
             LOGGER.error("Exception thrown executing query", e);
```

Comment:
This log message could be updated to specify a leaf partition query failed

Reply:
Update to include query id